TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php

```diff
@@ -93,21 +119,37 @@ public function listCommand($hostPattern = null)
             array_push($availableDomains, array(
                 'nodeName' => $domain->getSite()->getNodeName(),
                 'hostPattern' => $domain->getHostPattern(),
+                'scheme' => $domain->getScheme(),
+                'port' => $domain->getPort(),
                 'active' => $domain->getActive()
             ));
             if (strlen($domain->getSite()->getNodeName()) > $longestNodeName) {
                 $longestNodeName = strlen($domain->getSite()->getNodeName());
             }
-            if (strlen($domain->getHostPattern()) > $longestHostPattern) {
-                $longestHostPattern = strlen($domain->getHostPattern());
+            $url =
+                ($domain->getScheme() ? $domain->getScheme() . '://' : '') .
+                $domain->getHostPattern() .
+                ($domain->getPort() ? ':' . $domain->getPort() : '');
```

Comment:
What about inlining that "logic" to the Domain class and add some `Domain::getUrl()` method?